airbyte-integrations/connectors/destination-duckdb/destination_duckdb/destination.py

```diff
@@ -0,0 +1,172 @@
+from asyncio.log import logger
+from logging import getLogger
+
+from typing import Any, Iterable, Mapping
+
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.destinations import Destination
+from airbyte_cdk.models import AirbyteConnectionStatus, AirbyteMessage, ConfiguredAirbyteCatalog, DestinationSyncMode, Status, Type
+
+
+logger = getLogger("airbyte")
+
+
+class DestinationDuckdb(Destination):
+    @staticmethod
+    def _get_destination_path(destination_path: str) -> str:
+        """
+        Get a normalized version of the destination path.
+        Automatically append /local/ to the start of the path
+        """
+        if not destination_path.startswith("/local"):
+            destination_path = os.path.join("/local", destination_path)
+
+        destination_path = os.path.normpath(destination_path)
+        if not destination_path.startswith("/local"):
+            raise ValueError(
+                f"destination_path={destination_path} is not a valid path." "A valid path shall start with /local or no / prefix"
+            )
+
+        return destination_path
+
+    def write(
+        self, config: Mapping[str, Any], configured_catalog: ConfiguredAirbyteCatalog, input_messages: Iterable[AirbyteMessage]
+    ) -> Iterable[AirbyteMessage]:
+
+        """
+        Reads the input stream of messages, config, and catalog to write data to the destination.
+
+        This method returns an iterable (typically a generator of AirbyteMessages via yield) containing state messages received
+        in the input message stream. Outputting a state message means that every AirbyteRecordMessage which came before it has been
+        successfully persisted to the destination. This is used to ensure fault tolerance in the case that a sync fails before fully completing,
+        then the source is given the last state message output from this method as the starting point of the next sync.
+
+        :param config: dict of JSON configuration matching the configuration declared in spec.json
+        :param input_messages: The stream of input messages received from the source
+        :param configured_catalog: The Configured Catalog describing the schema of the data being received and how it should be persisted in the destination
+        :return: Iterable of AirbyteStateMessages wrapped in AirbyteMessage structs
+        """
+        streams = {s.stream.name for s in configured_catalog.streams}
+        logger.info(f"Starting write to DuckDB with {len(streams)} streams")
+
+        path = config.get("destination_path")
+        path = self._get_destination_path(path)
+        # check if file exists
+
+        if os.path.isfile(path):
+            con = duckdb.connect(database=path, read_only=False)
+            logger.info(f"Existing DuckDB file opened at {path}")
+
+        else:
+            con = duckdb.connect(database=path, read_only=False)
+            logger.info(f"Empty DuckDB file created at {path}")
```

Comment:
@sspaeti what this lines give to you?

Reply:
Good catch. Nothing, I removed that already locally. I thought I had to distinguish first if a file exists or not, but I learned I can always open with `raead_only=False`